### PR TITLE
K8SPG-270: Allow configuring pgBackRest options for scheduled backups

### PR DESCRIPTION
**Pull request by egegunes**
Add new `backrestOpts` field to scheduled backups and pass it to backup Job.

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| version-service | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/version-service.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/scheduled-backup.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/init-deploy.log) |
| upgrade | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/upgrade.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/scaling.log) |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/recreate.log) |
| affinity | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/affinity.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/smart-update.log) |
| monitoring | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/monitoring.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/self-healing.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/operator-self-healing.log) |
| clone-cluster | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/clone-cluster.log) |
| tls-check | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/tls-check.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/users.log) |
| ns-mode | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/ns-mode.log) |
| We run 16 out of 16|

commit: https://github.com/percona/percona-postgresql-operator/pull/351/commits/6f70f45ae1a134ed8d9e99fbc7b0b1288c401fff
image: `perconalab/percona-postgresql-operator:PR-351-6f70f45a-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-351-6f70f45a-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-351-6f70f45a-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-351-6f70f45a-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-351-6f70f45a-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-351-6f70f45a-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/demand-backup.log) |
| version-service | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/version-service.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/init-deploy.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/scheduled-backup.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/scaling.log) |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/recreate.log) |
| affinity | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/affinity.log) |
| upgrade | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/upgrade.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/monitoring.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/self-healing.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/operator-self-healing.log) |
| clone-cluster | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/clone-cluster.log) |
| tls-check | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/tls-check.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/users.log) |
| ns-mode | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/ns-mode.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/smart-update.log) |
| We run 16 out of 16|

commit: https://github.com/percona/percona-postgresql-operator/pull/351/commits/6f70f45ae1a134ed8d9e99fbc7b0b1288c401fff
image: `perconalab/percona-postgresql-operator:PR-351-6f70f45a-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-351-6f70f45a-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-351-6f70f45a-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-351-6f70f45a-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-351-6f70f45a-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-351-6f70f45a-pgo-deployer`

File: cmd/pgo-scheduler/scheduler/configmapcontroller.go [hunk 1]
```diff
@@ -60,11 +60,36 @@ func (c *Controller) onDelete(obj interface{}) {
 	c.Scheduler.DeleteSchedule(cm)
 }
 
+// onDelete is called when a configMap is updated
```
Comment (inelpandzic) [inline cmd/pgo-scheduler/scheduler/configmapcontroller.go:63]: You mean `onUpdate` is called..

**Review by inelpandzic — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/demand-backup.log) |
| upgrade | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/upgrade.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/init-deploy.log) |
| version-service | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/version-service.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/scaling.log) |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/recreate.log) |
| affinity | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/affinity.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/monitoring.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/self-healing.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/operator-self-healing.log) |
| clone-cluster | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/clone-cluster.log) |
| tls-check | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/tls-check.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/scheduled-backup.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/users.log) |
| ns-mode | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/ns-mode.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/6f70f45a/smart-update.log) |
| We run 16 out of 16|

commit: https://github.com/percona/percona-postgresql-operator/pull/351/commits/6f70f45ae1a134ed8d9e99fbc7b0b1288c401fff
image: `perconalab/percona-postgresql-operator:PR-351-6f70f45a-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-351-6f70f45a-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-351-6f70f45a-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-351-6f70f45a-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-351-6f70f45a-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-351-6f70f45a-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/3c8f4eee/demand-backup.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/3c8f4eee/scheduled-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/3c8f4eee/init-deploy.log) |
| upgrade | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/3c8f4eee/upgrade.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/3c8f4eee/version-service.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/3c8f4eee/smart-update.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/3c8f4eee/scaling.log) |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/3c8f4eee/recreate.log) |
| affinity | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/3c8f4eee/affinity.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/3c8f4eee/monitoring.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/3c8f4eee/self-healing.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/3c8f4eee/operator-self-healing.log) |
| clone-cluster | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/3c8f4eee/clone-cluster.log) |
| tls-check | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/3c8f4eee/tls-check.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/3c8f4eee/users.log) |
| ns-mode | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/3c8f4eee/ns-mode.log) |
| We run 16 out of 16|

commit: https://github.com/percona/percona-postgresql-operator/pull/351/commits/3c8f4eee84b3ab62ee82d9b2f5a11a5f96dd9981
image: `perconalab/percona-postgresql-operator:PR-351-3c8f4eee-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-351-3c8f4eee-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-351-3c8f4eee-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-351-3c8f4eee-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-351-3c8f4eee-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-351-3c8f4eee-pgo-deployer`

**Review by hors — APPROVED** [review]

File: deploy/cr.yaml [hunk 2]
```diff
@@ -126,6 +126,7 @@ spec:
         keep: 3
         type: full
         storage: local
+        backrestOpts: --annotation=source=scheduled-backup
```
Comment (hors) [inline deploy/cr.yaml:129]: @egegunes We need to comment it by default.

**Review by hors — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/5d406bf0/version-service.log) |
| upgrade | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/5d406bf0/upgrade.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/5d406bf0/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/5d406bf0/init-deploy.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/5d406bf0/scheduled-backup.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/5d406bf0/smart-update.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/5d406bf0/scaling.log) |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/5d406bf0/recreate.log) |
| affinity | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/5d406bf0/affinity.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/5d406bf0/monitoring.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/5d406bf0/self-healing.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/5d406bf0/operator-self-healing.log) |
| clone-cluster | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/5d406bf0/clone-cluster.log) |
| tls-check | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/5d406bf0/tls-check.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/5d406bf0/users.log) |
| ns-mode | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-351/5d406bf0/ns-mode.log) |
| We run 16 out of 16|

commit: https://github.com/percona/percona-postgresql-operator/pull/351/commits/5d406bf0d8c05dd625f189ad6bb0c49e8bd23bff
image: `perconalab/percona-postgresql-operator:PR-351-5d406bf0-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-351-5d406bf0-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-351-5d406bf0-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-351-5d406bf0-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-351-5d406bf0-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-351-5d406bf0-pgo-deployer`

**Review by hors — APPROVED** [review]

**Pull request merged**